### PR TITLE
Operation save-Ryan's-sanity has commenced.

### PR DESCRIPTION
**Pull request by ryanpbrewster**

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestingActor.scala [hunk 1]
```diff
@@ -108,14 +108,15 @@ class SlackIngestingActor @Inject() (
       case result =>
         val now = clock.now()
         val (nextIngestionAt, updatedStatus) = result match {
-          case Success(Some(_)) =>
+          case Success(Some(_)) => // TODO(ryan): talk to Léo, this branch is probably never executed
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestingActor.scala:111]: @leogrim I'm pretty sure this never happens, as `doIngest` has a `foldLeftUntil` that will keep running until `newLastMessageTimestamp.isEmpty` is true, so `newLastMessageTimestamp` will always be `None`. I don't think that's intended behavior.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackIngestingActor.scala:111]: I think that the `foldLeftUntil` in `doIngest` should return

```
(newLastMessageTimestamp orElse lastMessageTimestamp, newLastMessageTimestamp.isEmpty)
```

**Comment by leogrim** [comment]
Where is this coming from? Bug somewhere? Off-by-one error?

**Comment by ryanpbrewster** [comment]
I tried to set up a new user onboarding message that was triggered the first time an ingestion is processed, but I put the trigger in the code branch that is never executed so my messages weren't ever being sent.

**Pull request merged**